lib/execution_engine2/utils/catalog_cache.py

```diff
@@ -0,0 +1,89 @@
+from collections import defaultdict
+from typing import Dict
+
+from lib.installed_clients.CatalogClient import Catalog
+
+
+class CatalogCache:
+    """
+    Per call catalog cache used to speed up catalog lookups
+    Caches the "Method Version" and the "Job Resource Requirements"
+    """
+
+    def __init__(self, catalog: Catalog):
+        if not catalog:
+            raise ValueError("Please provide instance of catalog client")
+
+        self._catalog = catalog
+        self._method_version_cache = defaultdict(dict)
+        self._job_requirements_cache = defaultdict(dict)
+
+    def get_catalog(self) -> Catalog:
+        """Get the catalog client for this instance."""
+        return self._catalog
+
+    def get_method_version_cache(self) -> Dict:
+        """Get the _method_version_cache for this instance."""
+        return self._method_version_cache
+
+    def get_job_resources_cache(self) -> Dict:
+        """Get the _condor_resources_cache for this instance."""
+        return self._job_requirements_cache
+
+    def get_git_commit_version(self, method, service_ver=None) -> str:
+        """
+        If "service_ver" is "release|beta|dev", get git commit version for that version
+        if "service_ver" is a semantic version, get commit version for that semantic version
+        If "service_ver" is a git commit hash, see if that get commit is valid
+        self._catalog = Catalog(url=url, token=admin_token)
+        Convenience wrapper for verifying a git commit hash, or getting git commit hash from a tag
+        :param method: Method to look up
+        :param service_ver: Version to look up
+        :return: A git commit hash for the requested job
+        """
+
+        # Structure of cache
+        # { 'run_megahit' :
+        #   {
+        #       'dev' : 'cc91ddfe376f907aa56cfb3dd1b1b21cae8885z6', #Tag
+        #       '2.5.0' : 'cc91ddfe376f907aa56cfb3dd1b1b21cae8885z6', #Semantic
+        #       'cc91ddfe376f907aa56cfb3dd1b1b21cae8885z6' : 'cc91ddfe376f907aa56cfb3dd1b1b21cae8885z6' #vcs
+        #    }
+        # }
+        mv_cache = self.get_method_version_cache()
+        if method is None:
+            raise ValueError("Must provide a method to lookup")
+
+        if service_ver is None:
+            service_ver = "release"
+
+        # If not in the cache add it
+        if method not in mv_cache or service_ver not in mv_cache[method]:
+            module_name = method.split(".")[0]
```

Comment:
this line appears to be untested

Reply:
Looks tested to me https://codecov.io/gh/kbase/execution_engine2/src/16354d98d275b345a3976ac1d8f8ccd972ab0b98/lib/execution_engine2/utils/catalog_cache.py

Reply:
It's covered, but it's not tested - if I deleted the `split` call the tests would still pass.

Reply:
Ah you're right. GOod call added tests

Reply:
:+1: